src/components/dataTable/js/tableCellDirective.js

```diff
@@ -0,0 +1,213 @@
+  }
+}
+
+function MdTableOrderByController($element, $scope, $attrs, $compile, $mdUtil, $$mdSvgRegistry) {
+  this.$element = $element;
+  this.$scope = $scope;
+  this.$attrs = $attrs;
+  this.$compile = $compile;
+  this.$mdUtil = $mdUtil;
+  this.$$mdArrowUpIcon = $$mdSvgRegistry.mdArrowUp;
+
+  this.isActiveRegex = new RegExp('^-?' + this.$attrs.mdOrderBy + '$');
+
+  this.$attrs.$observe('mdOrderBy', function (orderBy) {
+    if (orderBy) {
+      this.enableSorting();
+    } else {
+      this.disableSorting();
+    }
+  }.bind(this));
+}
+
+MdTableOrderByController.prototype.initCtrl = function(tableCtrl, orderCtrl, cellCtrl) {
+  this.table = tableCtrl;
+  this.orderCtrl = orderCtrl;
+  this.cellCtrl = cellCtrl;
+
+  this.unregisterWatch = null;
+  this.clickListenerFn = this.updateSortOrder.bind(this);
+
+  // Watch for position changes to realign the sort icon.
+  this.$scope.$watch(function() {
+    return this.cellCtrl.$element[0].cellIndex;
+  }.bind(this), function() {
+    var icon = this.$element[0].querySelector('.md-sort-icon');
+
+    if (icon) {
+      if (this.$element.data('align') === 'right') {
+        this.$element.prepend(icon);
+      } else {
+        this.$element.append(icon);
+      }
+    }
+
+  }.bind(this));
+};
+
+MdTableOrderByController.prototype.isActive = function() {
+  // The isActiveRegex checks the headCtrl's order property.
+  // When the order direction is inverted, we prefix the order property
+  // with a minus.
+  return this.isActiveRegex.test(this.orderCtrl.value);
+};
+
+MdTableOrderByController.prototype.getDirection = function() {
+  if (this.isActive()) {
+    // When the current order value is prefixed with a minus, then its order
+    // is descendant.
+    return this.orderCtrl.value.charAt(0) === '-' ? 'md-desc' : 'md-asc';
+  } else {
+    return this.$mdUtil.parseAttributeBoolean(this.$attrs.mdDesc) ? 'md-desc' : 'md-asc';
+  }
+};
+
+MdTableOrderByController.prototype.compileIcon = function() {
+  return this.$compile(
+    '<md-icon class="md-sort-icon" ng-class="$mdOrderBy.getDirection()" ' +
+    '         md-svg-src="' + this.$$mdArrowUpIcon + '">'
+  )(this.$scope);
+};
+
+MdTableOrderByController.prototype.updateSortOrder = function() {
+  this.$scope.$applyAsync(function() {
+    var orderFilter = this.$attrs.mdOrderBy;
+    if (this.isActive()) {
+      var isDescendant = this.orderCtrl.value.charAt(0) === '-';
+      this.orderCtrl.setSortOrder(isDescendant ? orderFilter : '-' + orderFilter);
+    } else {
+      var isStaticDescendant = this.$mdUtil.parseAttributeBoolean(this.$attrs.mdDesc);
+      this.orderCtrl.setSortOrder(isStaticDescendant ? '-' + orderFilter : orderFilter);
+    }
+  }.bind(this));
+};
+
+MdTableOrderByController.prototype.enableSorting = function() {
+  this.$element
+    .addClass('md-sort')
+    .on('click', this.clickListenerFn);
+
+  if (!this.$element.children().length) {
+    this.$element.contents().wrap('<span>');
```

Comment:
It is a basic `jqLite` functionality, so it works for all Angular supported browsers.